meerk40t/core/elements.py

```diff
@@ -3258,21 +3272,477 @@ def element_grid(
+            ),
+            input_type=("elements", None),
+            output_type="elements",
+        )
+        def element_shape(
+            command,
+            channel,
+            _,
+            corners,
+            cx,
+            cy,
+            radius,
+            startangle=None,
+            inscribed=None,
+            radius_inner=None,
+            alternate_seq=None,
+            density=None,
+            data=None,
+            **kwargs,
+        ):
+            if corners is None:
+                raise SyntaxError
+            if corners <= 2:
+                if cx is None:
+                    cx = Length(0)
+                elif not cx.is_valid_length:
+                    raise SyntaxError("cx: " + _("This is not a valid length"))
+                if cy is None:
+                    cy = Length(0)
+                elif not cy.is_valid_length:
+                    raise SyntaxError("cy: " + _("This is not a valid length"))
+                cx = cx.value(
+                    ppi=1000, relative_length=bed_dim.bed_width * MILS_IN_MM
+                )
+                cy = cy.value(
+                    ppi=1000, relative_length=bed_dim.bed_width * MILS_IN_MM
+                )
+                if radius is None:
+                    radius = Length(0)
+                radius = radius.value(
+                    ppi=1000, relative_length=bed_dim.bed_width * MILS_IN_MM
+                )
+                if startangle is None:
+                    startangle = Angle.parse("0deg")
+
+                starpts = [(cx, cy)]
+                if corners==2:
+                    starpts += [(cx + cos(startangle.as_radians) * radius, cy + sin(startangle.as_radians) * radius)]
+
+            else:
+                if cx is None:
+                    raise SyntaxError(
+                        _(
+                            "Please provide at least one additional value (which will act as radius then)"
+                        )
+                    )
+                else:
+                    if not cx.is_valid_length:
+                        raise SyntaxError("cx: " + _("This is not a valid length"))
+
+                if cy is None:
+                    cy = Length(0)
+                else:
+                    if not cy.is_valid_length:
+                        raise SyntaxError("cy: " + _("This is not a valid length"))
+                # do we have something like 'polyshape 3 4cm' ? If yes, reassign the parameters
+                if radius is None:
+                    radius = cx
+                    cx = Length(0)
+                    cy = Length(0)
+                else:
+                    if not radius.is_valid_length:
+                        raise SyntaxError("radius: " + _("This is not a valid length"))
+
+                if startangle is None:
+                    startangle = Angle.parse("0deg")
+                if inscribed is None:  # Use circumscribed circle by default
```

Comment:
All this does is scale the radius (and maybe inner_radius?) so you don't need an explicit value for it, since the truthiness of None is False.